### PR TITLE
ISPN-14116 Create @TestForIssue annotation for the project

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-14116

### CODE REVIEW

File: commons-test/src/main/java/org/infinispan/commons/test/annotation/TestForIssue.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import java.lang.annotation.Target;
+
+/**
+ * A documentation annotation for notating what JIRA issue is being tested.
+ * <p>
+ * Copied from <a href="https://github.com/hibernate/hibernate-orm">Hibernate ORM project</a>.
+ *
+ * @author Steve Ebersole
+ */
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (tristantarrant) [inline commons-test/src/main/java/org/infinispan/commons/test/annotation/TestForIssue.java:15]: Does it really need to be `RUNTIME` ?

**Review by tristantarrant — CHANGES_REQUESTED** [review]

File: commons-test/src/main/java/org/infinispan/commons/test/annotation/TestForIssue.java (hunk 1 above)
Comment (fax4ever) [reply commons-test/src/main/java/org/infinispan/commons/test/annotation/TestForIssue.java]: I didn't ask me that question. Changed to `SOURCE` it seems to be fine.

**Comment by tristantarrant** [comment]
Merged, thanks

**Comment by fax4ever** [comment]
Thank you @tristantarrant

**Pull request merged**